garage/tf/models/gaussian_mlp_model2.py

```diff
@@ -0,0 +1,178 @@
+    :param adaptive_std: Is std a neural network. If False, it will be a
+        parameter.
+    :param std_share_network: Boolean for whether mean and std share the same
+        network.
+    :param std_hidden_sizes: List of sizes for the fully-connected layers
+        for std.
+    :param min_std: Whether to make sure that the std is at least some
+        threshold value, to avoid numerical issues.
+    :param max_std: Whether to make sure that the std is at most some
+        threshold value, to avoid numerical issues.
+    :param std_hidden_nonlinearity: Nonlinearity for each hidden layer in
+        the std network.
+    :param std_output_nonlinearity: Nonlinearity for output layer in
+        the std network.
+    :param std_parametrization: How the std should be parametrized. There
+        are a few options:
+        - exp: the logarithm of the std will be stored, and applied a
+            exponential transformation
+        - softplus: the std will be computed as log(1+exp(x))
+    :param hidden_nonlinearity: Nonlinearity used for each hidden layer.
+    :param output_nonlinearity: Nonlinearity for the output layer.
+    """
+
+    def __init__(self,
+                 output_dim,
+                 name=None,
+                 hidden_sizes=(32, 32),
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_hidden_sizes=(32, 32),
+                 std_parameterization='exp'):
+        # Network parameters
+        super().__init__(name)
+        self._hidden_sizes = hidden_sizes
+        self._output_dim = output_dim
+        self._learn_std = learn_std
+        self._adaptive_std = adaptive_std
+        self._std_share_network = std_share_network
+        self._std_hidden_sizes = std_hidden_sizes
+        self._min_std = min_std
+        self._max_std = max_std
+        self._std_parameterization = std_parameterization
+
+        # Tranform std arguments to parameterized space
+        self._init_std_param = None
+        self._min_std_param = None
+        self._max_std_param = None
+        if self._std_parameterization == 'exp':
+            self._init_std_param = np.log(init_std)
+            if min_std:
+                self._min_std_param = np.log(min_std)
+            if max_std:
+                self._max_std_param = np.log(max_std)
+        elif self._std_parameterization == 'softplus':
+            self._init_std_param = np.log(np.exp(init_std) - 1)
+            if min_std:
+                self._min_std_param = np.log(np.exp(min_std) - 1)
+            if max_std:
+                self._max_std_param = np.log(np.exp(max_std) - 1)
+        else:
+            raise NotImplementedError
+
+    def network_output_spec(self):
+        """Network output spec."""
+        return ['sample', 'log_std', 'distribution']
```

Comment:
please rename `distribution` to `dist`. abbr is more convenient here.

Reply:
Ok